stream_alert/rule_processor/rules_engine.py

```diff
@@ -14,26 +14,66 @@
 limitations under the License.
 """
 from copy import copy
-from datetime import timedelta
+from datetime import datetime, timedelta
 import os
 
 from stream_alert.rule_processor import LOGGER
 from stream_alert.rule_processor.threat_intel import StreamThreatIntel
 from stream_alert.shared import NORMALIZATION_KEY, resources
 from stream_alert.shared.alert import Alert
 from stream_alert.shared.rule import import_folders, Rule
+from stream_alert.shared.rule_table import RuleTable
 
 
 _IGNORE_KEYS = {StreamThreatIntel.IOC_KEY, NORMALIZATION_KEY}
 
 
 class RulesEngine(object):
     """Class to act as a rules engine that processes rules"""
+    _RULE_TABLE_LAST_REFRESH = None
+    _RULE_TABLE = None
+
     def __init__(self, config, *rule_paths):
         """Initialize a RulesEngine instance to cache a StreamThreatIntel instance."""
         self._threat_intel = StreamThreatIntel.load_from_config(config)
         self._required_outputs_set = resources.get_required_outputs()
         import_folders(*rule_paths)
+        self._load_rule_table(config)
+
+    @classmethod
+    def _load_rule_table(cls, config):
+        """Load and return a RuleTable class for communicating with the DynamoDB rule table
+
+        Args:
+            config (dict): Loaded configuration from 'conf/' directory
+
+        Returns:
+            rule_table.RuleTable: Loaded frontend for DynamoDB rules table
+        """
+        # Ensure the rules table is enabled
+        rt_config = config['global']['infrastructure']['rules_table']
+        if not rt_config.get('enabled', False):
+            return
+
+        now = datetime.utcnow()
+        refresh_delta = timedelta(minutes=rt_config.get('refresh_minutes', 10))
+
+        # The rule table will need 'refreshed' if 1) the table or last refresh time
+        # do not exist, or 2) if the refresh interval has been surpassed
+        needs_refresh = (not (cls._RULE_TABLE and cls._RULE_TABLE_LAST_REFRESH) or
+                         cls._RULE_TABLE_LAST_REFRESH + refresh_delta < now)
```

Comment:
If `cls._RULE_TABLE_LAST_REFRESH` is initialized to some date in the past (UNIX epoch, for example) instead of `None`, then I don't think you need this double conditional

Reply:
hey @austinbyers can you elaborate/show an example of what you mean? I'm not sure I'm understanding 🤔 are you suggesting that the class prop is instantiated with some default datetime value? ie here: 

https://github.com/airbnb/streamalert/pull/710/files/ebf8e8de667a5a7448cad6773fe3e4c898c42863#diff-30cc622fccfff0217d84af2ff5bcd7ffR33

Reply:
Yes, exactly. For example, the `Alert` class sets the last dispatched time to the epoch:

https://github.com/airbnb/streamalert/blob/89c5b63115f3af23681e109c3d8260cb46239e1d/stream_alert/shared/alert.py#L94-L95

So that we can use a single condition to pull alerts from a given time window:

https://github.com/airbnb/streamalert/blob/89c5b63115f3af23681e109c3d8260cb46239e1d/stream_alert/shared/alert_table.py#L114-L117

Reply:
👍 awesome thank you!